tests/test_file.py

```diff
@@ -75,23 +75,34 @@ def _test_file_put(self, replica: Replica, scheme: str, test_bucket: str, upload
 
     @testmode.integration
     def test_file_put_large(self):
-        tempdir = tempfile.gettempdir()
-        self._test_file_put_large(Replica.aws, "s3", self.s3_test_bucket, S3Uploader(tempdir, self.s3_test_bucket))
+
+        def upload_callable_creator(uploader_class: type):
+            def upload_callable(bucket: str, key: str):
+                tempdir = tempfile.gettempdir()
+                uploader = uploader_class(tempdir, bucket)
+
+                src_data = os.urandom(AWS_MIN_CHUNK_SIZE + 1)
+                with tempfile.NamedTemporaryFile(delete=True) as fh:
+                    fh.write(src_data)
+                    fh.flush()
+
+                    uploader.checksum_and_upload_file(fh.name, key, "text/plain")
+            return upload_callable
+
+        self._test_file_put_large(Replica.aws, self.s3_test_bucket, upload_callable_creator(S3Uploader))
         # There's no equivalent for GCP ... yet. :)
 
-    def _test_file_put_large(self, replica: Replica, scheme: str, test_bucket: str, uploader: Uploader):
+    def _test_file_put_large(self, replica: Replica, test_bucket: str, upload_func: typing.Callable[[str, str], None]):
```

Comment:
why do you need the None at the end?

Reply:
That's to indicate that the callable should return `None`.  I should fix L80 to reflect that as well...